app/models/message.rb

```diff
@@ -22,4 +22,18 @@ class Message < ApplicationRecord
   validates :text, presence: true
 
   scope :order_desc, -> { order(id: :desc) }
+
+  after_create :increment_unread
+
+  private
+
+  def target_match
```

Comment:
What do you think about adding an `enum` to know if a message is a text or is a match?
With the current approach, if the text changes, you will not be able to identify the matches.
Also, the `enum` will provide you with methods like `match?` to know if the message is a match.
WDYT?
Let me know if you want me to explain better the benefits of an enum.